### ISSUE TITLE
Jellyfin LDAP Authentication

### ISSUE THREAD

**Issue by NickSica** [jellyfin/jellyfin#6821]
### Please describe your bug

I was trying to connect Ombi to Jellyfin for user authentication and while the user would authenticate correctly on the backend, a null value error caused it to return a 401. I found the issue and my changes are in my [fork](https://github.com/NickSica/jellyfin/commit/4b2c40f71706d0b8e77925cb0103ae7b9fc65503). I am a bit wary about opening a pull request due to some behavior that was exhibited from taking the lines I ultimately changed out altogether. The behavior I noticed immediately was a broken the connection to jellyfin itself (on the server setup page in Ombi), but the authentication was working with existing Jellyfin users (The second log is from that). If all looks well, I can submit a pull request. 

Steps to Reproduce
1. Sign into Ombi
2. Connect to Jellyfin and Import Users
3. Try to login with an LDAP user

### Jellyfin Version

Other

### if other:

10.8.0

### Environment

```markdown
- OS: ArchLinux
- Virtualization: None
- Clients: Browser
- Browser: Chrome
- FFmpeg Version: 4.4-Jellyfin
- Playback Method: Direct Play
- Hardware Acceleration: none
- Plugins: LDAP
- Reverse Proxy: Nginx
- Base URL: none
- Networking: Host
- Storage: local
```

### Jellyfin logs

```shell
[2021-11-10 22:57:48.626 -05:00] [INF] [25] Emby.Server.Implementations.Session.SessionManager: Reissuing access token: "REDACTED"
[2021-11-10 22:57:48.630 -05:00] [ERR] [25] Jellyfin.Server.Middleware.ExceptionMiddleware: Error processing request. URL "POST" "/users/authenticatebyname".
System.ArgumentNullException: Value cannot be null. (Parameter 'deviceId')
   at Emby.Server.Implementations.Session.SessionManager.LogSessionActivity(String appName, String appVersion, String deviceId, String deviceName, String remoteEndPoint, User user) in /home/REDACTED/.cache/yay/jellyfin-git/src/jellyfin/Emby.Server.Implementations/Session/SessionManager.cs:line 280
   at Emby.Server.Implementations.Session.SessionManager.AuthenticateNewSessionInternal(AuthenticationRequest request, Boolean enforcePassword) in /home/REDACTED/.cache/yay/jellyfin-git/src/jellyfin/Emby.Server.Implementations/Session/SessionManager.cs:line 1501
   at Jellyfin.Api.Controllers.UserController.AuthenticateUserByName(AuthenticateUserByName request) in /home/REDACTED/.cache/yay/jellyfin-git/src/jellyfin/Jellyfin.Api/Controllers/UserController.cs:line 207
   at lambda_method348(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.AwaitableObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Jellyfin.Server.Middleware.ServerStartupMessageMiddleware.Invoke(HttpContext httpContext, IServerApplicationHost serverApplicationHost, ILocalizationManager localizationManager) in /home/REDACTED/.cache/yay/jellyfin-git/src/jellyfin/Jellyfin.Server/Middleware/ServerStartupMessageMiddleware.cs:line 41
   at Jellyfin.Server.Middleware.WebSocketHandlerMiddleware.Invoke(HttpContext httpContext, IWebSocketManager webSocketManager) in /home/REDACTED/.cache/yay/jellyfin-git/src/jellyfin/Jellyfin.Server/Middleware/WebSocketHandlerMiddleware.cs:line 33
   at Jellyfin.Server.Middleware.IpBasedAccessValidationMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager) in /home/REDACTED/.cache/yay/jellyfin-git/src/jellyfin/Jellyfin.Server/Middleware/IpBasedAccessValidationMiddleware.cs:line 47
   at Jellyfin.Server.Middleware.LanFilteringMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager, IServerConfigurationManager serverConfigurationManager) in /home/REDACTED/.cache/yay/jellyfin-git/src/jellyfin/Jellyfin.Server/Middleware/LanFilteringMiddleware.cs:line 42
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Server.Middleware.QueryStringDecodingMiddleware.Invoke(HttpContext httpContext) in /home/REDACTED/.cache/yay/jellyfin-git/src/jellyfin/Jellyfin.Server/Middleware/QueryStringDecodingMiddleware.cs:line 36
   at Swashbuckle.AspNetCore.ReDoc.ReDocMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Server.Middleware.RobotsRedirectionMiddleware.Invoke(HttpContext httpContext) in /home/REDACTED/.cache/yay/jellyfin-git/src/jellyfin/Jellyfin.Server/Middleware/RobotsRedirectionMiddleware.cs:line 44
   at Jellyfin.Server.Middleware.LegacyEmbyRouteRewriteMiddleware.Invoke(HttpContext httpContext) in /home/REDACTED/.cache/yay/jellyfin-git/src/jellyfin/Jellyfin.Server/Middleware/LegacyEmbyRouteRewriteMiddleware.cs:line 51
   at Jellyfin.Server.Middleware.ResponseTimeMiddleware.Invoke(HttpContext context) in /home/REDACTED/.cache/yay/jellyfin-git/src/jellyfin/Jellyfin.Server/Middleware/ResponseTimeMiddleware.cs:line 63
   at Jellyfin.Server.Middleware.ExceptionMiddleware.Invoke(HttpContext context) in /home/REDACTED/.cache/yay/jellyfin-git/src/jellyfin/Jellyfin.Server/Middleware/ExceptionMiddleware.cs:line 55



[2021-11-10 23:39:54.377 -05:00] [INF] [29] Emby.Server.Implementations.Session.SessionManager: Reissuing access token: "REDACTED"
[2021-11-10 23:39:54.566 -05:00] [WRN] [29] Jellyfin.Server.Middleware.ResponseTimeMiddleware: Slow HTTP Response from "https://jellyfin.REDACTED.com/users/authenticatebyname" to "REDACTED" in 0:00:00.8136586 with Status Code 200
[2021-11-10 23:40:04.153 -05:00] [ERR] [17] Jellyfin.Server.Middleware.ExceptionMiddleware: Error processing request. URL "GET" "/users".
System.ArgumentNullException: Value cannot be null. (Parameter 'value')
   at System.Security.Claims.Claim..ctor(String type, String value, String valueType, String issuer, String originalIssuer, ClaimsIdentity subject, String propertyKey, String propertyValue)
   at System.Security.Claims.Claim..ctor(String type, String value)
   at Jellyfin.Api.Auth.CustomAuthenticationHandler.HandleAuthenticateAsync() in /home/REDACTED/.cache/yay/jellyfin-git/src/jellyfin/Jellyfin.Api/Auth/CustomAuthenticationHandler.cs:line 54
   at Microsoft.AspNetCore.Authentication.AuthenticationHandler`1.AuthenticateAsync()
   at Microsoft.AspNetCore.Authentication.AuthenticationService.AuthenticateAsync(HttpContext context, String scheme)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Server.Middleware.RobotsRedirectionMiddleware.Invoke(HttpContext httpContext) in /home/REDACTED/.cache/yay/jellyfin-git/src/jellyfin/Jellyfin.Server/Middleware/RobotsRedirectionMiddleware.cs:line 44
   at Jellyfin.Server.Middleware.LegacyEmbyRouteRewriteMiddleware.Invoke(HttpContext httpContext) in /home/REDACTED/.cache/yay/jellyfin-git/src/jellyfin/Jellyfin.Server/Middleware/LegacyEmbyRouteRewriteMiddleware.cs:line 51
   at Jellyfin.Server.Middleware.ResponseTimeMiddleware.Invoke(HttpContext context) in /home/REDACTED/.cache/yay/jellyfin-git/src/jellyfin/Jellyfin.Server/Middleware/ResponseTimeMiddleware.cs:line 63
   at Jellyfin.Server.Middleware.ExceptionMiddleware.Invoke(HttpContext context) in /home/REDACTED/.cache/yay/jellyfin-git/src/jellyfin/Jellyfin.Server/Middleware/ExceptionMiddleware.cs:line 55
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by joshuaboniface** [comment]
A PR of the fix would be easier for everyone on the server team to analyze. I haven't noticed any strange issues on the 10.8.0 alpha1 with my Ombi instance but can do some further testing.

**Comment by NickSica** [comment]
> A PR of the fix would be easier for everyone on the server team to analyze. I haven't noticed any strange issues on the 10.8.0 alpha1 with my Ombi instance but can do some further testing.

Ok, I just created it. Strange that you aren't seeing the same behavior.

**Comment by crobibero** [comment]
Fixed in #6822
